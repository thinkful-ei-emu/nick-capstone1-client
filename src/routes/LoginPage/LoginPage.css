* {
  box-sizing: border-box;
}

 
.login-page {
  background-color: rgb(155, 163, 172);
  color: white;
  min-height: 200px;
  padding: 4%;
  margin: 10% 2%;
  border-radius: 10px;
}

.login-form {
  margin: 0 15%;
}

.form-row {
  display: flex;
  margin-bottom: 2%;
}

.form-row input {
  flex: 1;
}

.form-row label {
  margin-right: 2%;
}


.login-button {
  color: white;
  font-size: 18px;
  min-height: 30px;
  width: auto;
  background-color: rgb(207, 55, 55);
  border: 1px solid rgb(207, 55, 55);
  border-radius: 10px;
  margin: 4% 0;
}

@media only screen and (min-width: 600px){
  .login-page {
    background-color: rgb(155, 163, 172);
    color: white;
    min-height: 200px;
    margin: 10% 20%;
    border-radius: 10px;
  }
}

@media only screen and (min-width: 1000px){
  .login-page {
    background-color: rgb(155, 163, 172);
    color: white;
    min-height: 300px;
    margin: 6% 20%;
    border-radius: 10px;
  }
  .login-form {
    margin: 0 20%;
  }
}

