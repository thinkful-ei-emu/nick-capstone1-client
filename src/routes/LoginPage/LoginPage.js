import React from 'react';
import { Link } from 'react-router-dom';
import LoginForm from '../../components/LoginForm/LoginForm'
import './LoginPage.css'
import HeaderContext from '../../contexts/HeaderContext';


class LoginPage extends React.Component {
   static defaultProps = {
     history: {
       push: () => {},
     },
   }

   static contextType = HeaderContext

   handleLoginSuccess = (token) => {
     const { history } = this.props;
     this.context.setAuthToken(token)
     history.push('/posts')
   }

  render(){
    return (
      <>
      <Link to={'/posts'}>
        <h1 className='title-header'>BandBridge</h1>
        </Link>
      <section className='login-page'> 
      <h1 className='login-header'>Login</h1>
        <LoginForm 
        onLoginSuccess={this.handleLoginSuccess}
        />
        </section>
      </>
    )
  }  
}

export default LoginPage;