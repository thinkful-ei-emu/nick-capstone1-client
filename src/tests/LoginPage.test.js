import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from '../routes/LoginPage/LoginPage';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';




it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<MemoryRouter><LoginPage /></MemoryRouter>, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders UI as expected', () => {
  const tree = renderer
      .create(<MemoryRouter><LoginPage /></MemoryRouter>)
      .toJSON();
      expect(tree).toMatchSnapshot();
})