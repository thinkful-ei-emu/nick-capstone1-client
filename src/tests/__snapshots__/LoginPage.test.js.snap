// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
Array [
  <a
    href="/posts"
    onClick={[Function]}
  >
    <h1
      className="title-header"
    >
      BandBridge
    </h1>
  </a>,
  <section
    className="login-page"
  >
    <h1
      className="login-header"
    >
      Login
    </h1>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <div
        role="alert"
      />
      <div
        className="user_name form-row"
      >
        <label
          htmlFor="login-user"
        >
          User name: 
        </label>
        <input
          className="input"
          id="login-user"
          name="user_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="password form-row"
      >
        <label
          htmlFor="login-password"
        >
          Password: 
        </label>
        <input
          className="input"
          id="login-password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="login-button"
        type="submit"
      >
        Login
      </button>
    </form>
  </section>,
]
`;
